api.planx.uk/modules/flows/updateTemplatedFlow/customMerge.test.ts

```diff
@@ -0,0 +1,289 @@
+      },
+    },
+    Content: {
+      type: 250,
+      data: {
+        content: "<h1>Test</h1><p>This is some example content</p>",
+      },
+    },
+    ContentNew: {
+      type: 250,
+      data: {
+        content: "NEW !",
+      },
+    },
+  });
+});
+
+it("handles templated folders where all nodes have been removed", () => {
+  const sourceTemplate: FlowGraph = {
+    _root: {
+      edges: ["Folder"],
+    },
+    Folder: {
+      type: 300,
+      edges: ["Content"],
+      data: {
+        text: "Folder",
+        isTemplatedNode: true,
+        templatedNodeInstructions:
+          "You can empty this folder if you don't want to ask these questions",
+      },
+    },
+    Content: {
+      type: 250,
+      data: {
+        content: "<h1>Test</h1><p>This is some example content</p>",
+      },
+    },
+  };
+
+  const templatedFlowEdits: Flow["data"] = {
+    Folder: {
+      edges: [],
+    },
+  };
+
+  expect(customMerge(sourceTemplate, templatedFlowEdits)).toEqual({
+    _root: {
+      edges: ["Folder"],
+    },
+    Folder: {
+      type: 300,
+      edges: [],
+      data: {
+        text: "Folder",
+        isTemplatedNode: true,
+        templatedNodeInstructions:
+          "You can empty this folder if you don't want to ask these questions",
+      },
+    },
+  });
+});
+
+it("handles templated upload and label nodes where rules have been removed, added, and re-ordered", () => {
+  const sourceTemplate: FlowGraph = {
+    _root: {
+      edges: ["UploadAndLabel"],
+    },
+    UploadAndLabel: {
+      type: 145,
+      data: {
+        title: "Upload and label",
+        fileTypes: [
+          {
+            name: "Site plan",
+            fn: "sitePlan.proposed",
+            rule: {
+              condition: "AlwaysRequired",
+            },
+          },
+          {
+            name: "Design and access statement",
+            fn: "designAndAccessStatement",
+            rule: {
+              condition: "AlwaysRecommended",
+            },
+          },
+        ],
+        hideDropZone: false,
+        tags: [],
+        isTemplatedNode: true,
+        templatedNodeInstructions: "Set your local file requirements",
+      },
+    },
+  };
+
+  // Remove design & access, add heritage statement, re-order so site plan is last
+  const templatedFlowEdits: Flow["data"] = {
+    UploadAndLabel: {
+      data: {
+        fileTypes: [
+          {
+            fn: "heritageStatement",
+            name: "Heritage statement",
+            rule: {
+              condition: "AlwaysRequired",
+            },
+          },
+          {
+            fn: "sitePlan.proposed",
+            name: "Site plan",
+            rule: {
+              condition: "AlwaysRequired",
+            },
+            moreInformation: {
+              info: "<p>It shows where the work is happening</p>",
+            },
+          },
+        ],
+      },
+    },
+  };
+
+  expect(customMerge(sourceTemplate, templatedFlowEdits)).toEqual({
```

Comment:
By introducing a `customMerge` we now get added benefit of easier to unit test - we can essentially mock mini source flows and their edits and see what the reconciliation produces